proof_ipa.go

```diff
@@ -102,16 +104,47 @@ func (sd StateDiff) Copy() StateDiff {
 	ret := make(StateDiff, len(sd))
 	for i := range sd {
 		copy(ret[i].Stem[:], sd[i].Stem[:])
-		ret[i].SuffixDiffs = make([]SuffixStateDiff, len(sd[i].SuffixDiffs))
-		for j := range sd[i].SuffixDiffs {
-			ret[i].SuffixDiffs[j].Suffix = sd[i].SuffixDiffs[j].Suffix
-			if sd[i].SuffixDiffs[j].CurrentValue != nil {
-				ret[i].SuffixDiffs[j].CurrentValue = &[32]byte{}
-				copy((*ret[i].SuffixDiffs[j].CurrentValue)[:], (*sd[i].SuffixDiffs[j].CurrentValue)[:])
+
+		ret[i].UpdatedSuffixes = make([]byte, len(sd[i].UpdatedSuffixes))
+		copy(ret[i].UpdatedSuffixes, sd[i].UpdatedSuffixes)
+		ret[i].ReadSuffixes = make([]byte, len(sd[i].ReadSuffixes))
+		copy(ret[i].ReadSuffixes, sd[i].ReadSuffixes)
+		ret[i].InsertedSuffixes = make([]byte, len(sd[i].InsertedSuffixes))
+		copy(ret[i].InsertedSuffixes, sd[i].InsertedSuffixes)
+		ret[i].UntouchedSuffixes = make([]byte, len(sd[i].UntouchedSuffixes))
+		copy(ret[i].UntouchedSuffixes, sd[i].UntouchedSuffixes)
+
+		ret[i].UpdatedCurrent = make([][]byte, len(sd[i].UpdatedCurrent))
+		for j := range sd[i].UpdatedCurrent {
+			if len(sd[i].UpdatedCurrent[j]) == 0 {
+
+			} else {
+				copy(ret[i].UpdatedCurrent[j], sd[i].UpdatedCurrent[j])
```

Comment:
Same comments as mentioned in [this other PR](https://github.com/ethereum/go-verkle/pull/441/files#r1686968659). Same applies in below cases in this method.